pages/en/supported-networks/cosmos.mdx

```diff
@@ -0,0 +1,246 @@
+
+```bash
+$ graph codegen
+```
+
+Once the mappings are ready, the subgraph needs to be built. This step will highlight any errors the manifest or the mappings might have. A subgraph needs to build successfully in order to be deployed to the Graph Node. It can be done using the `build` CLI command:
+
+```bash
+$ graph build
+```
+
+## Deploying a Cosmos subgraph
+
+When the subgraph builds successfully, it is time to deploy it to a Graph Node instance for indexing. Cosmos subgraphs can be deployed to any Graph Node on version `>=v0.26.0`.
+
+The first thing to do is create the subgraph in the node, this only has to be done once via the `create` CLI command:
+
+```bash
+Local deployment:
+$ graph create subgraph-name --node <graph-node-url>
+
+Hosted service deployment:
+$ graph create subgraph-name --node https://api.thegraph.com/deploy/
+```
+
+After the subgraph has been created, it can be deployed by using the graph `deploy` CLI command:
+
+```bash
+Local deployment:
+$ graph deploy subgraph-name --ipfs <ipfs-url> --node <graph-node-url>
+
+Hosted service deployment:
+$ graph deploy subgraph-name --ipfs https://api.thegraph.com/ipfs/ --node https://api.thegraph.com/deploy/
+```
+
+Once your subgraph has been deployed, it will be indexed by Graph Node. You can check its progress by querying the subgraph itself at the exposed endpoint.
+
+## Querying a Cosmos subgraph
+
+The GraphQL endpoint for Cosmos subgraphs is determined by the schema definition, with the existing API interface. Please visit the [GraphQL API documentation](https://thegraph.com/docs/en/developer/graphql-api/) for more information.
+
+## Supported Cosmos Blockchains
+
+### Cosmos Hub
+
+##### What is Cosmos Hub?
+
+The [Cosmos Hub blockchain](https://hub.cosmos.network/) is the first blockchain in the [Cosmos](https://cosmos.network/) ecosystem. You can visit the [official documentation](https://docs.cosmos.network/) for more information.
+
+##### Subgraph Dependencies
+
+[graph-cli](https://github.com/graphprotocol/graph-cli) is a CLI tool to build and deploy subgraphs, version `>=0.30.0` is required in order to work with Cosmos Hub subgraphs.
+
+[graph-ts](https://github.com/graphprotocol/graph-ts) is a library of subgraph-specific types, version `>=0.27.0` is required in order to work with Cosmos Hub subgraphs.
+
+##### Networks
+
+Cosmos Hub mainnet is `cosmoshub-4`. Cosmos Hub current testnet is `theta-testnet-001`. <br/>Other Cosmos Hub networks, i.e. `cosmoshub-3`, are halted, therefore no data is provided for them.
+
+##### Example Subgraphs
+
+Here are some example subgraphs for reference:
+
+[Cosmos Hub Block Filtering Example](https://github.com/graphprotocol/example-subgraph/tree/cosmos-block-example)
+
+[Cosmos Hub Validator Rewards Example](https://github.com/graphprotocol/example-subgraph/tree/cosmos-event-example)
+
+[Cosmos Hub Validator Delegations Example](https://github.com/graphprotocol/example-subgraph/tree/cosmos-transaction-example)
```

Comment:
We can remove `Cosmos Hub` from the link descriptions.

Reply:
Also, we need to update the branch names (in our fork and upstream).

Reply:
If we use templating I guess the names are ok right ?